pilot/pkg/networking/core/v1alpha3/networkfilter.go

```diff
@@ -116,16 +101,63 @@ func buildDeprecatedTCPProxyFilter(clusterName string, addr string) (*listener.F
 	return tcpFilter, nil
 }
 
-// buildOutboundNetworkFilters generates TCP proxy network filter for outbound connections. In addition, it generates
-// protocol specific filters (e.g., Mongo filter)
-// this function constructs deprecated_v1 routes, until the filter chain match is ready
-func buildOutboundNetworkFilters(env *model.Environment, node *model.Proxy, clusterName string,
-	deprecatedTCPFilterMatchAddress string, port *model.Port) []listener.Filter {
+// buildStatPrefix builds a stat prefix by concatenating the given route
+// destinations and port.
+func buildStatPrefix(routes []*networking.RouteDestination, port int, push *model.PushContext) string {
+	var routeNames []string
+	for _, route := range routes {
+		var routeName string
+		service := push.ServiceByHostname[model.Hostname(route.Destination.Host)]
+		if route.Destination.Subset != "" {
+			if service != nil && len(service.Ports) == 1 {
+				routeName = fmt.Sprintf("%s:%s:%d", route.Destination.Host, route.Destination.Subset, service.Ports[0].Port)
+			} else {
+				routeName = fmt.Sprintf("%s:%s", route.Destination.Host, route.Destination.Subset)
+			}
+		} else {
+			if service != nil && len(service.Ports) == 1 {
+				routeName = fmt.Sprintf("%s:%d", route.Destination.Host, service.Ports[0].Port)
+			} else {
+				routeName = route.Destination.Host
+			}
+		}
+		routeNames = append(routeNames, routeName)
+	}
+	sort.Strings(routeNames)
+	return model.BuildSubsetKey(model.TrafficDirectionOutbound, "", model.Hostname(strings.Join(routeNames[:], "_")), port)
+}
```

Comment:
this whole thing is very kludgey. Figure out a different stat prefix. There is no need to have a stat prefix with all destination and subset names.. it will be too big and probably cause other issues. 
I suggest using the virtualservice name.namespace as the stat prefix for weighted clusters.

Reply:
Addressed in c164e2d.